src/common/context/shared-context.js

```diff
@@ -17,5 +18,7 @@ export class SharedContext {
     } catch (err) {
       warn('An error occured while setting SharedContext', err)
     }
+
+    this.timeKeeper = TimeKeeper.getTimeKeeperByAgentIdentifier(this.sharedContext.agentIdentifier)
```

Comment:
This is fine but just wanted to mention that the way this shared context module currently works is it extracts pointers from the parent on L14-17. If `timeKeeper` was in the model on L4-6 it would automatically grab it from the feature and you wouldn't need to define it / fetch it here.  If that's a good or bad pattern is a different discussion point for a different time 😆

Reply:
Honestly, I'm beginning to hate the word `context`. There is room for improvement here, the inheritance tree between our features and other code like the harvester is weird and confusing.